Add trouble shooting about how to transpile node_modules (#883)
babel/babel-loader: README.md

> This README is for babel-loader v8 + Babel v7

> > For docs with Babel v6 see [7.x branch](https://github.com/babel/babel-loader/tree/7.x) 

Reviewer: ```suggestion
> If you are using Babel v6, see the [7.x branch](https://github.com/babel/babel-loader/tree/7.x) docs
```

Author: What about "If you are using legacy Babel v6"? I really want to drop this line as Babel 8 is to be released 😄

Reviewer: Sure

You can also speed up babel-loader by as much as 2x by using the `cacheDirectory` option. This will cache transformations to the filesystem.

### Some files in my node_modules are not transpiled for IE 11

Although we typically recommend not compiling `node_modules`, you may need to when using libraries that do not support IE 11.

> to compile them for IE 11 compatibility.

Reviewer: Remove this line

> For this, we suggest passing an `exclude` function for maintainability. You can also use negative lookahead regex as suggested [here](https://github.com/webpack/webpack/issues/2031#issuecomment-294706065).

Reviewer: Maybe something like?

```suggestion
For this, you can either use a combination of `test` and `not`, or [pass a function](https://webpack.js.org/configuration/module/#condition) to your `exclude` option. You can also use negative lookahead regex as suggested [here](https://github.com/webpack/webpack/issues/2031#issuecomment-294706065).
```

```javascript
{
    test: /\.m?js$/,
    exclude: {
      test: /node_modules/, // Exclude libraries in node_modules ...
      not: [

>         // Except for a few of them that needs to be transpiled because use modern syntax

Reviewer: ```suggestion
        // Except for a few of them that needs to be transpiled because they use modern syntax
```

You can also speed up babel-loader by as much as 2x by using the `cacheDirectory` option. This will cache transformations to the filesystem.

> ### Some files in my node_modules are not transpiled to IE 11

Reviewer: ```suggestion
### Some files in my node_modules are not transpiled for IE 11
```

> Although we recommend to exclude `node_modules`, there are cases when some libraries do not support IE 11 and you may need

Reviewer: ```suggestion
Although we typically recommend not compiling `node_modules`, you may need to when using libraries that do not support IE 11.
```

(The following line "to compile them for IE 11 compatibility." can be dropped).

to compile them for IE 11 compatibility.

> In this case we suggest use an `exclude` function for maintainability. You can also use negative lookahead regex as suggested [here](https://github.com/webpack/webpack/issues/2031#issuecomment-294706065).

Reviewer: ```suggestion
For this, we suggest passing an `exclude` function for maintainability. You can also use negative lookahead regex as suggested [here](https://github.com/webpack/webpack/issues/2031#issuecomment-294706065).
```

```javascript
{
    test: /\.m?js$/,

>     exclude: function excludeBabelLoader(modulePath) {
>       if (/node_modules/.test(modulePath)) {
>         if (
>           /node_modules[\\/]d3-array[\\/]/.test(modulePath) // compile d3-array to IE 11
>           || /node_modules[\\/](unfetch|d3-scale)[\\/]/.test(modulePath) // compile unfetch and d3-scale to IE 11
>           || /node_modules[\\/]@hapi[\\/]joi-date[\\/]/.test(modulePath) // compile @hapi/joi-date to IE 11
>         ) {
>           return false;
>         }
>         // Exclude other libraries in node_modules
>         return true;
>       }
>       // Compile every others not in node_modules
>       return false;
>     },

Reviewer 2: ```suggestion
    exclude: {
      test: /node_modules/, // Exclude libraries in node_modules ...
      not: [
        // Except for a few of them that needs to be transpiled because use modern syntax
        /unfetch/,
        /d3-array|d3-scale/,
        /@hapi[\\/]joi-date/,
      ]
    },
```

This is maybe more reasable?

Reviewer: Definitely more readable. Maybe lead with this, and mention for more advanced situations use exclude as a function (link to webpack exclude docs)?
